File: src/OpenFleet.Domain/Entities/Inspection.cs
using OpenFleet.Domain.Enums;

namespace OpenFleet.Domain.Entities;

public class Inspection : BaseEntity
{
    public Guid? VehicleId { get; set; }
    public Guid? AssetId { get; set; }
    public Guid InspectorUserId { get; set; }
    public DateTime InspectedAt { get; set; }
    public InspectionStatus Status { get; set; } = InspectionStatus.Passed;
    public string Notes { get; set; } = string.Empty;
    public Guid? GeneratedWorkOrderId { get; set; }

    public Vehicle? Vehicle { get; set; }
    public Asset? Asset { get; set; }
    public User InspectorUser { get; set; } = null!;
    public WorkOrder? GeneratedWorkOrder { get; set; }
}

File: src/OpenFleet.Web/src/types/inspection.ts
export type InspectionStatus = 'Passed' | 'Failed' | 'NeedsReview'

export interface InspectionResponse {
  id: string
  vehicleId: string | null
  vehicleDescription: string | null
  assetId: string | null
  assetDescription: string | null
  inspectorUserId: string
  inspectorName: string
  inspectedAt: string
  status: InspectionStatus
  notes: string
  generatedWorkOrderId: string | null
  createdAt: string
  updatedAt: string
}

export interface CreateInspectionRequest {
  vehicleId?: string
  assetId?: string
  inspectorUserId: string
  inspectedAt: string
  status: InspectionStatus
  notes?: string
}
